### PR TITLE
Add `color` token group using the new internal palette

### PR DESCRIPTION
**Pull request by aaronccasanova**
Part of #6920 

This PR introduces a new `color` token group that utilizes a collection of colors from our new internal palette. Our revised naming convention (prefixing token names with the token group name) ensures that there are no conflicts with the existing `colors` tokens. Which means the new values can temporarily coexist with the legacy values. In addition to being able to ship these semantic alias tokens now, we can start migrating to the new tokens in Polaris and in Web, before removing legacy tokens in the Polaris v11 release.

Example output:
Screenshot 2023-02-10 at 3 47 00 PM

### CODE REVIEW

**Review by lgriffee — APPROVED** [review]
Looks good! Only thing I caught was something Github brought up: 
Screenshot 2023-02-02 at 1 09 14 PM
Do we need to add `color` here in addition to `colors`?

File: polaris-tokens/src/token-groups/color.ts [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+  [TokenName in ColorTokenName as `color-${TokenName}`]: string;
+};
+
+export const color: {
+  [TokenName in ColorTokenName as `color-${TokenName}`]: MetadataProperties;
+} = {
+  'color-bg-backdrop': {
+    value: 'rgba(0, 0, 0, 0.5)',
+    description: '',
+  },
```
Comment (alex-page) [inline polaris-tokens/src/token-groups/color.ts]: Should this be a token?

File: polaris-tokens/src/token-groups/color.ts [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+  'color-bg': {
+    value: palettes.gray[50],
+    description: '',
+  },
+  'color-bg-overlay': {
+    value: 'rgba(255, 255, 255, 0.5)',
+    description: '',
+  },
```
Comment (alex-page) [inline polaris-tokens/src/token-groups/color.ts]: Should this be a token?

File: polaris-tokens/src/token-groups/color.ts (hunk 1 above)
Comment (aaronccasanova) [reply polaris-tokens/src/token-groups/color.ts]: Spoke with Joe and he's okay with removing this token. There are only 5 instances in Web and 1 in Polaris.

File: polaris-tokens/src/token-groups/color.ts (hunk 2 above)
Comment (aaronccasanova) [reply polaris-tokens/src/token-groups/color.ts]: Spoke with Joe and he's okay with removing this token. There are only 3 instances in Web and 4 in Polaris.

File: polaris-tokens/src/token-groups/color.ts (hunk 1 above)
Comment (alex-page) [reply polaris-tokens/src/token-groups/color.ts]: Nice. I think we remove it and add a comment where the hardcoded value is used that it does `x` custom thing.

File: polaris-tokens/src/token-groups/color.ts (hunk 2 above)
Comment (alex-page) [reply polaris-tokens/src/token-groups/color.ts]: Nice. I think we remove it and add a comment where the hardcoded value is used that it does `x` custom thing.

File: polaris-tokens/src/token-groups/color.ts (hunk 1 above)
Comment (aaronccasanova) [reply polaris-tokens/src/token-groups/color.ts]: Removed https://github.com/Shopify/polaris/pull/8246/commits/ec4dd4511f64c08236b94a4164efc6f14b4759b5

File: polaris-tokens/src/token-groups/color.ts (hunk 2 above)
Comment (aaronccasanova) [reply polaris-tokens/src/token-groups/color.ts]: Removed https://github.com/Shopify/polaris/pull/8246/commits/ec4dd4511f64c08236b94a4164efc6f14b4759b5

**Review by sam-b-rose — APPROVED** [review]
🎨 💯

File: .changeset/breezy-glasses-cry.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-tokens': minor
+---
+
+Added a `color` token group with new internal palettes
```
Comment (aaronccasanova) [inline .changeset/breezy-glasses-cry.md]:
```suggestion
Added a `color` token group using the new internal palette
```

**Review by alex-page — APPROVED** [review]
Looks good to me. Definitely need @heyjoethomas to take a look before shipping.

**Review by heyjoethomas — APPROVED** [review]
🚀

**Pull request merged**